builtin/providers/aws/resource_aws_api_gateway_base_path_mapping_test.go

```diff
@@ -0,0 +1,137 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+	//"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSAPIGatewayBasePath_basic(t *testing.T) {
+	var conf apigateway.BasePathMapping
+	name := "test-api-gateway-domain-cert0.com"
```

Comment:
That's actually rather awkward here, since API gateway seems to verify that the cert is valid for the domain.

Perhaps we have no option here but to talk to the Go crypto libraries directly in these tests and issue some fresh certs each time the tests run. Shouldn't be hard to do, but kinda annoying to have that much logic in a test. :grimacing:

We can at least keep the private key "hard-coded", which is the part that tends to be slowest due to all of the crunching needed to produce a good amount of randomness.